binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java

```diff
@@ -154,9 +156,15 @@ static List<String> beanNamesForMatchingSerdes(ConfigurableApplicationContext co
 				BeanDefinition beanDefinition = context.getBeanFactory().getMergedBeanDefinition(beanName);
 				ResolvableType serdeBeanGeneric = beanDefinition.getResolvableType().getGeneric(0);
 				if (LOG.isDebugEnabled()) {
-					LOG.debug("Found matching Serde<" + serdeBeanGeneric.getType() + "> under beanName=" + beanName);
+					LOG.debug("Found matching Serde<" + serdeBeanGeneric.getType() + "> with beanName=" + beanName);
+				}
+				Class<?> serdeGenericRawClazz = serdeBeanGeneric.getRawClass();
+				if (serdeGenericRawClazz != null) {
+					matchingSerdes.add(new SerdeWithSpecificityScore(calculateScore(targetType, serdeBeanGeneric), beanName));
+				}
+				else if (LOG.isDebugEnabled()) {
+					LOG.debug("Discarding raw or wildcard match Serde<" + serdeBeanGeneric.getType() + "> with beanName=" + beanName);
```

Comment:
Just adding a few debug statements to help understand why things do / not match.